### PR TITLE
Add MDMMotionCurveReversed for reversing timing curves.

### PR DESCRIPTION
**Pull request by jverkoey**

### CODE REVIEW

**Comment by jverkoey** [comment]
Will add tests before landing.

**Comment by jverkoey** [comment]
Tests added. Ready for review.

File: src/MDMMotionCurve.h [hunk 1]
```diff
@@ -94,6 +94,11 @@ FOUNDATION_EXTERN MDMMotionCurve MDMMotionCurveMakeSpring(float mass, float tens
     NS_SWIFT_NAME(MotionCurveMakeSpring(mass:tension:friction:));
 // clang-format on
 
+// clang-format off
+FOUNDATION_EXTERN MDMMotionCurve MDMMotionCurveReversed(MDMMotionCurve timingCurve)
```
Comment (romoore) [inline src/MDMMotionCurve.h]: Documentation to explain that it only works for Bezier curves?

File: src/MDMMotionCurve.m [hunk 2]
```diff
@@ -31,3 +31,14 @@ MDMMotionCurve MDMMotionCurveFromTimingFunction(CAMediaTimingFunction *timingFun
...
 }
+
+MDMMotionCurve MDMMotionCurveReversed(MDMMotionCurve timingCurve) {
+  MDMMotionCurve reversed = timingCurve;
+  if (timingCurve.type == MDMMotionCurveTypeBezier) {
+    reversed.data[0] = 1 - timingCurve.data[2];
+    reversed.data[1] = 1 - timingCurve.data[3];
+    reversed.data[2] = 1 - timingCurve.data[0];
+    reversed.data[3] = 1 - timingCurve.data[1];
+  }
```
Comment (romoore) [inline src/MDMMotionCurve.m]: What other curve types are there? Maybe it should be called `MDMMotionBezierCurveReversed` if it only supports the `.bezier` type?

File: tests/unit/MDMMotionCurveTests.swift [hunk 3]
```diff
@@ -43,4 +43,13 @@ class MDMMotionCurveTests: XCTestCase {
...
   }
+
+  func testReversedBezierCurve() {
+    let curve = MotionCurveMakeBezier(p1x: 0.1, p1y: 0.2, p2x: 0.3, p2y: 0.4)
+    let reversed = MotionCurveReversed(fromTimingCurve: curve)
+    XCTAssertEqualWithAccuracy(curve.data.0, 1 - reversed.data.2, accuracy: 0.001)
+    XCTAssertEqualWithAccuracy(curve.data.1, 1 - reversed.data.3, accuracy: 0.001)
+    XCTAssertEqualWithAccuracy(curve.data.2, 1 - reversed.data.0, accuracy: 0.001)
+    XCTAssertEqualWithAccuracy(curve.data.3, 1 - reversed.data.1, accuracy: 0.001)
+  }
```
Comment (romoore) [inline tests/unit/MDMMotionCurveTests.swift:54]: One more test to confirm that reversing a reversed curve restores the original?

**Review by romoore — CHANGES_REQUESTED** [review]
One more test would be nice, but I think it's most important to document that only beziers are supported or add support for non-bezier curves.

File: tests/unit/MDMMotionCurveTests.swift (hunk 3 above)
Comment (jverkoey) [reply tests/unit/MDMMotionCurveTests.swift:54]: Done.

File: src/MDMMotionCurve.m (hunk 2 above)
Comment (jverkoey) [reply src/MDMMotionCurve.m]: Changed the API name.

File: src/MDMMotionCurve.h (hunk 1 above)
Comment (jverkoey) [reply src/MDMMotionCurve.h]: Done.

**Comment by jverkoey** [comment]
Solid feedback @romoore, thank you!

**Review by romoore — APPROVED** [review]

**Pull request merged**